crates/next-core/src/next_client/context.rs

```diff
@@ -350,6 +350,13 @@ pub async fn get_client_module_options_context(
         enable_postcss_transform,
         side_effect_free_packages: next_config.optimize_package_imports().owned().await?,
         keep_last_successful_parse: next_mode.is_development(),
+        enable_intermediate_tree_shaking: matches!(
```

Comment:
I see why you made it a separate option from `tree_shaking_mode`, but we will want a better name. Maybe something like `unused_export_(removal|tree_shaking|skipping)`

Reply:
Thank you for the suggestion! I wasn't sure about the name.